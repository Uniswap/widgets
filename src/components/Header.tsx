import { largeIconCss } from 'icons'
import { PropsWithChildren, ReactElement } from 'react'
import styled from 'styled-components/macro'
import { ThemedText } from 'theme'

import Row from './Row'

const HeaderRow = styled(Row)`
  height: 1.5rem;
  margin: 0.5rem 0.75rem 1rem;
  ${largeIconCss}
`

export interface HeaderProps {
  title?: ReactElement
}

export default function Header({ title, children }: PropsWithChildren<HeaderProps>) {
  return (
    <HeaderRow iconSize={1.2} flex align="center" data-testid="header-container">
      {title && (
        <Row gap={0.5} data-testid="header-title">
          <ThemedText.Subhead1>{title}</ThemedText.Subhead1>
        </Row>
      )}
      {children && (
        <Row gap={1} data-testid="header-children">
          {children}
        </Row>
      )}
    </HeaderRow>
  )
}
