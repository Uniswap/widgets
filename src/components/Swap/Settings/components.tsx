import { Info } from 'icons'
import { ReactNode } from 'react'
import styled, { AnyStyledComponent, css } from 'styled-components/macro'
import { ThemedText } from 'theme'

import Row from '../../Row'
import Tooltip from '../../Tooltip'

export const optionCss = (selected: boolean) => css`
  border: 1px solid ${({ theme }) => (selected ? theme.active : '')};
  border-radius: ${({ theme }) => theme.borderRadius.small}rem;
  color: ${({ theme }) => theme.primary} !important;
  display: grid;
  grid-gap: 0.25rem;
  padding: calc(0.75rem - 1px) 0.625rem;

  :enabled {
    border: 1px solid ${({ theme }) => (selected ? theme.active : theme.outline)};
  }

  :enabled:hover {
    border-color: ${({ theme }) => theme.onHover(selected ? theme.active : theme.outline)};
  }

  :enabled:focus-within {
    border-color: ${({ theme }) => theme.active};
  }
`

export function value(Value: AnyStyledComponent) {
  return styled(Value)<{ selected?: boolean; cursor?: string }>`
    cursor: ${({ cursor }) => cursor ?? 'pointer'};
  `
}

interface LabelProps {
  name: ReactNode
  tooltip?: ReactNode
}

export function Label({ name, tooltip }: LabelProps) {
  return (
    <Row gap={0.5} justify="flex-start" flex align="center">
      <ThemedText.Subhead2>{name}</ThemedText.Subhead2>
      {tooltip && (
        <Tooltip placement="top" contained icon={Info} iconProps={{ style: { height: '100%' } }}>
          <ThemedText.Caption>{tooltip}</ThemedText.Caption>
        </Tooltip>
      )}
    </Row>
  )
}
