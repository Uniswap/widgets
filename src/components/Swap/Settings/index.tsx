import { Trans } from '@lingui/macro'
import useScrollbar from 'hooks/useScrollbar'
import { Settings as SettingsIcon } from 'icons'
import { useResetAtom } from 'jotai/utils'
import React, { useState } from 'react'
import { settingsAtom } from 'state/settings'
import styled from 'styled-components/macro'
import { ThemedText } from 'theme'

import { IconButton, TextButton } from '../../Button'
import Column from '../../Column'
import Dialog, { Header } from '../../Dialog'
import { BoundaryProvider } from '../../Popover'
import MaxSlippageSelect from './MaxSlippageSelect'
import TransactionTtlInput from './TransactionTtlInput'

export function SettingsDialog() {
  const [boundary, setBoundary] = useState<HTMLDivElement | null>(null)
  const scrollbar = useScrollbar(boundary, { padded: true })
  const resetSettings = useResetAtom(settingsAtom)
  return (
    <>
      <Header title={<Trans>Settings</Trans>} ruled>
        <TextButton onClick={resetSettings}>
          <ThemedText.ButtonSmall>
            <Trans>Reset</Trans>
          </ThemedText.ButtonSmall>
        </TextButton>
      </Header>
      <Column gap={1} style={{ paddingTop: '1em' }} ref={setBoundary} padded css={scrollbar}>
        <BoundaryProvider value={boundary}>
          <MaxSlippageSelect />
          <TransactionTtlInput />
        </BoundaryProvider>
      </Column>
    </>
  )
}

const SettingsButton = styled(IconButton)<{ hover: boolean }>`
  ${SettingsIcon} {
    transform: ${({ hover }) => hover && 'rotate(45deg)'};
    transition: ${({ hover }) => hover && 'transform 0.25s'};
    will-change: transform;
  }
`

export default function Settings({ disabled }: { disabled?: boolean }) {
  const [open, setOpen] = useState(false)
  const [hover, setHover] = useState(false)
  return (
    <>
      <SettingsButton
        disabled={disabled}
        hover={hover}
        onClick={() => setOpen(true)}
        onMouseEnter={() => setHover(true)}
        onMouseLeave={() => setHover(false)}
        icon={SettingsIcon}
      />
      {open && (
        <Dialog color="module" onClose={() => setOpen(false)}>
          <SettingsDialog />
        </Dialog>
      )}
    </>
  )
}
