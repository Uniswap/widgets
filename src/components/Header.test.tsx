import { Trans } from '@lingui/macro'
import { renderComponent } from 'test'

import Header from './Header'
import Settings from './Swap/Settings'

describe('Header', () => {
  it('renders empty', () => {
    const component = renderComponent(<Header />)
    expect(component.queryByTestId('header-container')).toBeTruthy()
    expect(component.queryByTestId('header-title')).toBeNull()
    expect(component.queryByTestId('header-children')).toBeNull()
  })

  it('renders with title', () => {
    const component = renderComponent(<Header title={<Trans>Test Title</Trans>} />)
    expect(component.queryByTestId('header-container')).toBeTruthy()
    expect(component.queryByTestId('header-title')).toBeTruthy()
    expect(component.getByText('Test Title')).toBeTruthy()
    expect(component.queryByTestId('header-children')).toBeNull()
  })

  it('renders with children (actual values)', () => {
    const component = renderComponent(
      <Header title={<Trans>Swap</Trans>}>
        <Settings />
      </Header>
    )
    expect(component.queryByTestId('header-container')).toBeTruthy()
    expect(component.queryByTestId('header-title')).toBeTruthy()
    expect(component.getByText('Swap')).toBeTruthy()
    expect(component.queryByTestId('header-children')).toBeTruthy()
    expect(component.queryByTestId('settings-button')).toBeTruthy()
    expect(component).toMatchSnapshot()
  })
})
