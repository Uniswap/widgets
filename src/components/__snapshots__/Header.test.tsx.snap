// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders with children (actual values) 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div />
      <div
        style="isolation: isolate;"
      >
        <div
          class="Row-sc-1nzvhrh-0 Header__HeaderRow-sc-1a8935c-0 iYjZBm jTGDHe"
          data-testid="header-container"
        >
          <div
            class="Row-sc-1nzvhrh-0 lkMSnc"
            data-testid="header-title"
          >
            <div
              class="type__TextWrapper-sc-16386l-0 bRHiaw subhead subhead-1 css-mxz2od"
            >
              Swap
            </div>
          </div>
          <div
            class="Row-sc-1nzvhrh-0 jeLwvy"
            data-testid="header-children"
          >
            <div>
              <div
                class="Popover__Reference-sc-1liex6z-1 hplZxv"
              >
                <button
                  class="Button__BaseButton-sc-1soikk5-0 Button__transparentButton-sc-1soikk5-2 Button__StyledIconButton-sc-1soikk5-3 gccXME jBWboA cguYJw Settings__SettingsButton-sc-1hvqnx5-1 ewOqAq"
                  data-testid="settings-button"
                >
                  <svg
                    class="icons-sc-lekdau-0 iSRwlR"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="3"
                    />
                    <path
                      d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="Popover__PopoverContainer-sc-1liex6z-0 blItxa"
                data-testid="popover-container"
                style="position: absolute; left: 0px; top: 0px;"
              >
                <div
                  class="Column-sc-1ul9eki-0 Settings__SettingsColumn-sc-1hvqnx5-0 dsMBcY kNwgA-d"
                >
                  <div
                    class="Column-sc-1ul9eki-0 jowptB"
                  >
                    <div
                      class="Column-sc-1ul9eki-0 kcFdvT"
                    >
                      <div
                        class="Row-sc-1nzvhrh-0 gpwEJV"
                        style="cursor: pointer;"
                      >
                        <div
                          class="Row-sc-1nzvhrh-0 jIdqqq"
                        >
                          <div
                            class="type__TextWrapper-sc-16386l-0 jukanU subhead subhead-2 css-e17nie"
                          >
                            Max slippage
                          </div>
                          <div
                            class="Popover__Reference-sc-1liex6z-1 hplZxv"
                          >
                            <button
                              class="Button__BaseButton-sc-1soikk5-0 Button__transparentButton-sc-1soikk5-2 Button__StyledIconButton-sc-1soikk5-3 gccXME jBWboA cguYJw Tooltip__IconTooltip-sc-tsxpgp-1 fPAwBP"
                              style="outline: none;"
                              tabindex="-1"
                            >
                              <svg
                                class="icons-sc-lekdau-0 iSRwlR"
                                fill="none"
                                height="24"
                                stroke="currentColor"
                                stroke-linecap="round"
                                stroke-linejoin="round"
                                stroke-width="2"
                                style="height: 100%;"
                                viewBox="0 0 24 24"
                                width="24"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <circle
                                  cx="12"
                                  cy="12"
                                  r="10"
                                />
                                <line
                                  x1="12"
                                  x2="12"
                                  y1="16"
                                  y2="12"
                                />
                                <line
                                  x1="12"
                                  x2="12.01"
                                  y1="8"
                                  y2="8"
                                />
                              </svg>
                            </button>
                          </div>
                        </div>
                        <div
                          class="Row-sc-1nzvhrh-0 dZQqZC"
                        >
                          <div
                            class="Expando__IconPrefix-sc-yzkwmi-8 cffTyD"
                          >
                            Auto
                          </div>
                          <button
                            class="Button__BaseButton-sc-1soikk5-0 Button__transparentButton-sc-1soikk5-2 Button__StyledIconButton-sc-1soikk5-3 gccXME jBWboA cguYJw"
                            color="secondary"
                          />
                        </div>
                      </div>
                      <div
                        class="Column-sc-1ul9eki-0 Expando__ExpandoColumn-sc-yzkwmi-4 kcFdvT fcnTlu"
                      >
                        <div
                          class="Column-sc-1ul9eki-0 Expando__InnerColumn-sc-yzkwmi-5 Expando___StyledInnerColumn2-sc-yzkwmi-6 hUSZUI cMnglD gyFZzL"
                        >
                          <div
                            class="Row-sc-1nzvhrh-0 MaxSlippageSelect__ExpandoContentRow-sc-9adwbi-2 cfUpfx gnYxGO"
                          >
                            <button
                              class="Button__BaseButton-sc-1soikk5-0 Button__transparentButton-sc-1soikk5-2 MaxSlippageSelect__Button-sc-9adwbi-0 gccXME cQhCRX iOvcmB"
                              data-testid="auto-slippage"
                            >
                              <div
                                class="Row-sc-1nzvhrh-0 BEuty"
                              >
                                <div
                                  class="type__TextWrapper-sc-16386l-0 fMszLd button button-medium css-17g4kvt"
                                >
                                  Auto
                                </div>
                                <div
                                  class="icons__LargeWrapper-sc-lekdau-1 gyYXVl"
                                  color="active"
                                />
                              </div>
                            </button>
                            <button
                              class="Button__BaseButton-sc-1soikk5-0 MaxSlippageSelect__Custom-sc-9adwbi-1 gccXME ePlutd"
                              data-testid="custom-slippage"
                              style="outline: none;"
                              tabindex="-1"
                            >
                              <div
                                class="Row-sc-1nzvhrh-0 eSaSDE"
                              >
                                <div
                                  class="Row-sc-1nzvhrh-0 fUVNpP"
                                >
                                  <input
                                    autocomplete="off"
                                    autocorrect="off"
                                    class="Input-sc-1e35ws5-0 qhvlL"
                                    data-testid="input-slippage"
                                    inputmode="decimal"
                                    maxlength="10"
                                    minlength="1"
                                    pattern="^[0-9]*[.,]?[0-9]*$"
                                    placeholder="0.10"
                                    size="4"
                                    spellcheck="false"
                                    type="text"
                                    value=""
                                  />
                                  %
                                </div>
                                <div
                                  class="icons__LargeWrapper-sc-lekdau-1 gyYXVl"
                                  color="hint"
                                />
                              </div>
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <hr
                    class="Rule-sc-1568p9j-0 RJyeV"
                  />
                  <div
                    class="Column-sc-1ul9eki-0 jowptB"
                  >
                    <div
                      class="Column-sc-1ul9eki-0 kcFdvT"
                    >
                      <div
                        class="Row-sc-1nzvhrh-0 kSeFcF"
                        style="cursor: pointer;"
                      >
                        <div
                          class="Row-sc-1nzvhrh-0 jIdqqq"
                        >
                          <div
                            class="type__TextWrapper-sc-16386l-0 jukanU subhead subhead-2 css-e17nie"
                          >
                            Transaction deadline
                          </div>
                          <div
                            class="Popover__Reference-sc-1liex6z-1 hplZxv"
                          >
                            <button
                              class="Button__BaseButton-sc-1soikk5-0 Button__transparentButton-sc-1soikk5-2 Button__StyledIconButton-sc-1soikk5-3 gccXME jBWboA cguYJw Tooltip__IconTooltip-sc-tsxpgp-1 fPAwBP"
                              style="outline: none;"
                              tabindex="-1"
                            >
                              <svg
                                class="icons-sc-lekdau-0 iSRwlR"
                                fill="none"
                                height="24"
                                stroke="currentColor"
                                stroke-linecap="round"
                                stroke-linejoin="round"
                                stroke-width="2"
                                style="height: 100%;"
                                viewBox="0 0 24 24"
                                width="24"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <circle
                                  cx="12"
                                  cy="12"
                                  r="10"
                                />
                                <line
                                  x1="12"
                                  x2="12"
                                  y1="16"
                                  y2="12"
                                />
                                <line
                                  x1="12"
                                  x2="12.01"
                                  y1="8"
                                  y2="8"
                                />
                              </svg>
                            </button>
                          </div>
                        </div>
                        <div
                          class="Row-sc-1nzvhrh-0 dZQqZC"
                        >
                          <div
                            class="Expando__IconPrefix-sc-yzkwmi-8 cffTyD"
                          >
                            <div
                              class="TransactionTtlInput__TtlValue-sc-1fem02o-2 fyRhSr"
                            >
                              30
                              m
                            </div>
                          </div>
                          <button
                            class="Button__BaseButton-sc-1soikk5-0 Button__transparentButton-sc-1soikk5-2 Button__StyledIconButton-sc-1soikk5-3 gccXME jBWboA cguYJw"
                            color="secondary"
                          />
                        </div>
                      </div>
                      <div
                        class="Column-sc-1ul9eki-0 Expando__ExpandoColumn-sc-yzkwmi-4 kcFdvT fcnTlu"
                      >
                        <div
                          class="Column-sc-1ul9eki-0 Expando__InnerColumn-sc-yzkwmi-5 Expando___StyledInnerColumn2-sc-yzkwmi-6 hUSZUI cHlJMW gyFZzL"
                        >
                          <div
                            class="Row-sc-1nzvhrh-0 TransactionTtlInput__InputContainer-sc-1fem02o-1 gQrhhv eMmnPF"
                          >
                            <div
                              class="Row-sc-1nzvhrh-0 TransactionTtlInput__Input-sc-1fem02o-0 fcddMn itTgvP"
                            >
                              <input
                                autocomplete="off"
                                autocorrect="off"
                                class="Input-sc-1e35ws5-0 qhvlL"
                                inputmode="decimal"
                                maxlength="10"
                                minlength="1"
                                pattern="^[0-9]*$"
                                placeholder="30"
                                spellcheck="false"
                                type="text"
                                value=""
                              />
                              <div
                                class="type__TextWrapper-sc-16386l-0 hnjWoB body body-2 css-ijlljt"
                              >
                                minutes
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="Popover__PopoverContainer-sc-1liex6z-0 blItxa"
                    data-testid="popover-container"
                    style="position: absolute; left: 0px; top: 0px;"
                  >
                    <div
                      class="type__TextWrapper-sc-16386l-0 ihjAsA caption css-1up6pmj"
                    >
                      Your transaction will revert if the price changes unfavorably by more than this percentage.
                    </div>
                    <div
                      class="Popover__Arrow-sc-1liex6z-2 dQIWmB arrow-"
                      style="position: absolute;"
                    />
                  </div>
                  <div
                    class="Popover__PopoverContainer-sc-1liex6z-0 blItxa"
                    data-testid="popover-container"
                    style="position: absolute; left: 0px; top: 0px;"
                  >
                    <div
                      class="type__TextWrapper-sc-16386l-0 ihjAsA caption css-1up6pmj"
                    >
                      Your transaction will revert if it has been pending for longer than this period of time.
                    </div>
                    <div
                      class="Popover__Arrow-sc-1liex6z-2 dQIWmB arrow-"
                      style="position: absolute;"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div />
    <div
      style="isolation: isolate;"
    >
      <div
        class="Row-sc-1nzvhrh-0 Header__HeaderRow-sc-1a8935c-0 iYjZBm jTGDHe"
        data-testid="header-container"
      >
        <div
          class="Row-sc-1nzvhrh-0 lkMSnc"
          data-testid="header-title"
        >
          <div
            class="type__TextWrapper-sc-16386l-0 bRHiaw subhead subhead-1 css-mxz2od"
          >
            Swap
          </div>
        </div>
        <div
          class="Row-sc-1nzvhrh-0 jeLwvy"
          data-testid="header-children"
        >
          <div>
            <div
              class="Popover__Reference-sc-1liex6z-1 hplZxv"
            >
              <button
                class="Button__BaseButton-sc-1soikk5-0 Button__transparentButton-sc-1soikk5-2 Button__StyledIconButton-sc-1soikk5-3 gccXME jBWboA cguYJw Settings__SettingsButton-sc-1hvqnx5-1 ewOqAq"
                data-testid="settings-button"
              >
                <svg
                  class="icons-sc-lekdau-0 iSRwlR"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="3"
                  />
                  <path
                    d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
                  />
                </svg>
              </button>
            </div>
            <div
              class="Popover__PopoverContainer-sc-1liex6z-0 blItxa"
              data-testid="popover-container"
              style="position: absolute; left: 0px; top: 0px;"
            >
              <div
                class="Column-sc-1ul9eki-0 Settings__SettingsColumn-sc-1hvqnx5-0 dsMBcY kNwgA-d"
              >
                <div
                  class="Column-sc-1ul9eki-0 jowptB"
                >
                  <div
                    class="Column-sc-1ul9eki-0 kcFdvT"
                  >
                    <div
                      class="Row-sc-1nzvhrh-0 gpwEJV"
                      style="cursor: pointer;"
                    >
                      <div
                        class="Row-sc-1nzvhrh-0 jIdqqq"
                      >
                        <div
                          class="type__TextWrapper-sc-16386l-0 jukanU subhead subhead-2 css-e17nie"
                        >
                          Max slippage
                        </div>
                        <div
                          class="Popover__Reference-sc-1liex6z-1 hplZxv"
                        >
                          <button
                            class="Button__BaseButton-sc-1soikk5-0 Button__transparentButton-sc-1soikk5-2 Button__StyledIconButton-sc-1soikk5-3 gccXME jBWboA cguYJw Tooltip__IconTooltip-sc-tsxpgp-1 fPAwBP"
                            style="outline: none;"
                            tabindex="-1"
                          >
                            <svg
                              class="icons-sc-lekdau-0 iSRwlR"
                              fill="none"
                              height="24"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                              style="height: 100%;"
                              viewBox="0 0 24 24"
                              width="24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <circle
                                cx="12"
                                cy="12"
                                r="10"
                              />
                              <line
                                x1="12"
                                x2="12"
                                y1="16"
                                y2="12"
                              />
                              <line
                                x1="12"
                                x2="12.01"
                                y1="8"
                                y2="8"
                              />
                            </svg>
                          </button>
                        </div>
                      </div>
                      <div
                        class="Row-sc-1nzvhrh-0 dZQqZC"
                      >
                        <div
                          class="Expando__IconPrefix-sc-yzkwmi-8 cffTyD"
                        >
                          Auto
                        </div>
                        <button
                          class="Button__BaseButton-sc-1soikk5-0 Button__transparentButton-sc-1soikk5-2 Button__StyledIconButton-sc-1soikk5-3 gccXME jBWboA cguYJw"
                          color="secondary"
                        />
                      </div>
                    </div>
                    <div
                      class="Column-sc-1ul9eki-0 Expando__ExpandoColumn-sc-yzkwmi-4 kcFdvT fcnTlu"
                    >
                      <div
                        class="Column-sc-1ul9eki-0 Expando__InnerColumn-sc-yzkwmi-5 Expando___StyledInnerColumn2-sc-yzkwmi-6 hUSZUI cMnglD gyFZzL"
                      >
                        <div
                          class="Row-sc-1nzvhrh-0 MaxSlippageSelect__ExpandoContentRow-sc-9adwbi-2 cfUpfx gnYxGO"
                        >
                          <button
                            class="Button__BaseButton-sc-1soikk5-0 Button__transparentButton-sc-1soikk5-2 MaxSlippageSelect__Button-sc-9adwbi-0 gccXME cQhCRX iOvcmB"
                            data-testid="auto-slippage"
                          >
                            <div
                              class="Row-sc-1nzvhrh-0 BEuty"
                            >
                              <div
                                class="type__TextWrapper-sc-16386l-0 fMszLd button button-medium css-17g4kvt"
                              >
                                Auto
                              </div>
                              <div
                                class="icons__LargeWrapper-sc-lekdau-1 gyYXVl"
                                color="active"
                              />
                            </div>
                          </button>
                          <button
                            class="Button__BaseButton-sc-1soikk5-0 MaxSlippageSelect__Custom-sc-9adwbi-1 gccXME ePlutd"
                            data-testid="custom-slippage"
                            style="outline: none;"
                            tabindex="-1"
                          >
                            <div
                              class="Row-sc-1nzvhrh-0 eSaSDE"
                            >
                              <div
                                class="Row-sc-1nzvhrh-0 fUVNpP"
                              >
                                <input
                                  autocomplete="off"
                                  autocorrect="off"
                                  class="Input-sc-1e35ws5-0 qhvlL"
                                  data-testid="input-slippage"
                                  inputmode="decimal"
                                  maxlength="10"
                                  minlength="1"
                                  pattern="^[0-9]*[.,]?[0-9]*$"
                                  placeholder="0.10"
                                  size="4"
                                  spellcheck="false"
                                  type="text"
                                  value=""
                                />
                                %
                              </div>
                              <div
                                class="icons__LargeWrapper-sc-lekdau-1 gyYXVl"
                                color="hint"
                              />
                            </div>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <hr
                  class="Rule-sc-1568p9j-0 RJyeV"
                />
                <div
                  class="Column-sc-1ul9eki-0 jowptB"
                >
                  <div
                    class="Column-sc-1ul9eki-0 kcFdvT"
                  >
                    <div
                      class="Row-sc-1nzvhrh-0 kSeFcF"
                      style="cursor: pointer;"
                    >
                      <div
                        class="Row-sc-1nzvhrh-0 jIdqqq"
                      >
                        <div
                          class="type__TextWrapper-sc-16386l-0 jukanU subhead subhead-2 css-e17nie"
                        >
                          Transaction deadline
                        </div>
                        <div
                          class="Popover__Reference-sc-1liex6z-1 hplZxv"
                        >
                          <button
                            class="Button__BaseButton-sc-1soikk5-0 Button__transparentButton-sc-1soikk5-2 Button__StyledIconButton-sc-1soikk5-3 gccXME jBWboA cguYJw Tooltip__IconTooltip-sc-tsxpgp-1 fPAwBP"
                            style="outline: none;"
                            tabindex="-1"
                          >
                            <svg
                              class="icons-sc-lekdau-0 iSRwlR"
                              fill="none"
                              height="24"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                              style="height: 100%;"
                              viewBox="0 0 24 24"
                              width="24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <circle
                                cx="12"
                                cy="12"
                                r="10"
                              />
                              <line
                                x1="12"
                                x2="12"
                                y1="16"
                                y2="12"
                              />
                              <line
                                x1="12"
                                x2="12.01"
                                y1="8"
                                y2="8"
                              />
                            </svg>
                          </button>
                        </div>
                      </div>
                      <div
                        class="Row-sc-1nzvhrh-0 dZQqZC"
                      >
                        <div
                          class="Expando__IconPrefix-sc-yzkwmi-8 cffTyD"
                        >
                          <div
                            class="TransactionTtlInput__TtlValue-sc-1fem02o-2 fyRhSr"
                          >
                            30
                            m
                          </div>
                        </div>
                        <button
                          class="Button__BaseButton-sc-1soikk5-0 Button__transparentButton-sc-1soikk5-2 Button__StyledIconButton-sc-1soikk5-3 gccXME jBWboA cguYJw"
                          color="secondary"
                        />
                      </div>
                    </div>
                    <div
                      class="Column-sc-1ul9eki-0 Expando__ExpandoColumn-sc-yzkwmi-4 kcFdvT fcnTlu"
                    >
                      <div
                        class="Column-sc-1ul9eki-0 Expando__InnerColumn-sc-yzkwmi-5 Expando___StyledInnerColumn2-sc-yzkwmi-6 hUSZUI cHlJMW gyFZzL"
                      >
                        <div
                          class="Row-sc-1nzvhrh-0 TransactionTtlInput__InputContainer-sc-1fem02o-1 gQrhhv eMmnPF"
                        >
                          <div
                            class="Row-sc-1nzvhrh-0 TransactionTtlInput__Input-sc-1fem02o-0 fcddMn itTgvP"
                          >
                            <input
                              autocomplete="off"
                              autocorrect="off"
                              class="Input-sc-1e35ws5-0 qhvlL"
                              inputmode="decimal"
                              maxlength="10"
                              minlength="1"
                              pattern="^[0-9]*$"
                              placeholder="30"
                              spellcheck="false"
                              type="text"
                              value=""
                            />
                            <div
                              class="type__TextWrapper-sc-16386l-0 hnjWoB body body-2 css-ijlljt"
                            >
                              minutes
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="Popover__PopoverContainer-sc-1liex6z-0 blItxa"
                  data-testid="popover-container"
                  style="position: absolute; left: 0px; top: 0px;"
                >
                  <div
                    class="type__TextWrapper-sc-16386l-0 ihjAsA caption css-1up6pmj"
                  >
                    Your transaction will revert if the price changes unfavorably by more than this percentage.
                  </div>
                  <div
                    class="Popover__Arrow-sc-1liex6z-2 dQIWmB arrow-"
                    style="position: absolute;"
                  />
                </div>
                <div
                  class="Popover__PopoverContainer-sc-1liex6z-0 blItxa"
                  data-testid="popover-container"
                  style="position: absolute; left: 0px; top: 0px;"
                >
                  <div
                    class="type__TextWrapper-sc-16386l-0 ihjAsA caption css-1up6pmj"
                  >
                    Your transaction will revert if it has been pending for longer than this period of time.
                  </div>
                  <div
                    class="Popover__Arrow-sc-1liex6z-2 dQIWmB arrow-"
                    style="position: absolute;"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
